src/views/LoginSuccess.vue

```diff
@@ -66,21 +66,37 @@ export default class LoginSuccess extends Vue {
 
     private result?: LoginResult;
 
+    private render: boolean = false;
+    private retrievalComplete: boolean = false;
+
     private async mounted() {
+        const currentWalletInfo = await WalletStore.Instance.get(this.keyguardResult.keyId);
+
+        // if there is a WalletInfo for the id already, use it
+        if (currentWalletInfo && currentWalletInfo.label) {
+            this.walletLabel = currentWalletInfo.label;
+            if (currentWalletInfo.accounts) {
+                this.accounts = currentWalletInfo.accounts;
+            }
+        }
+
         // The Keyguard always returns (at least) one derived Address,
         // thus we can already create a complete WalletInfo object that
         // can be displayed while waiting for the network.
+        // before adding, make sure it is not set already
```

Comment:
The existing comment uses proper capitalization and punctuation. Please do, too.